sylvia/tests/remote.rs

```diff
@@ -1,62 +1,379 @@
+            &self,
+            ctx: ExecCtx<Self::QueryC>,
+            value: Self::CounterT,
+        ) -> Result<Response<Self::ExecC>, Self::Error> {
+            self.counter
+                .update(ctx.deps.storage, |current_value| -> StdResult<_> {
+                    Ok(current_value + value)
+                })?;
+            Ok(Response::new())
+        }
+
+        fn count(&self, ctx: QueryCtx<Self::QueryC>) -> Result<Self::CounterT, Self::Error> {
+            let count = self.counter.load(ctx.deps.storage)?;
+            Ok(count)
+        }
     }
 }
 
-#[contract]
-#[sv::messages(some_interface)]
-impl SomeContract {
-    #[allow(clippy::new_without_default)]
-    pub fn new() -> Self {
-        Self
+pub mod unsigned_contract {
+    use cosmwasm_std::{Response, StdError, StdResult};
+    use cw_storage_plus::Item;
+    use sylvia::{
+        contract,
+        types::{ExecCtx, InstantiateCtx, QueryCtx},
+    };
+
+    use crate::{counter::Counter, ExampleMsg, ExampleQuery};
+
+    pub struct UnsignedContract<'a> {
+        counter: Item<'a, u32>,
     }
 
-    #[sv::msg(instantiate)]
-    pub fn instantiate(&self, _ctx: InstantiateCtx) -> StdResult<Response> {
-        Ok(Response::new())
+    #[contract]
+    #[sv::messages(crate::counter<u32>)]
+    #[sv::custom(msg=ExampleMsg, query=ExampleQuery)]
+    impl UnsignedContract<'_> {
+        #[allow(clippy::new_without_default)]
+        pub fn new() -> Self {
+            Self {
+                counter: Item::new("counter"),
+            }
+        }
+
+        #[sv::msg(instantiate)]
+        pub fn instantiate(
+            &self,
+            ctx: InstantiateCtx<ExampleQuery>,
+        ) -> StdResult<Response<ExampleMsg>> {
+            self.counter.save(ctx.deps.storage, &0)?;
+            Ok(Response::new())
+        }
+    }
+
+    impl Counter for UnsignedContract<'_> {
+        type Error = StdError;
+        type ExecC = ExampleMsg;
+        type QueryC = ExampleQuery;
+        type CounterT = u32;
+
+        fn add(
+            &self,
+            ctx: ExecCtx<Self::QueryC>,
+            value: Self::CounterT,
+        ) -> Result<Response<Self::ExecC>, Self::Error> {
+            self.counter
+                .update(ctx.deps.storage, |current_value| -> StdResult<_> {
+                    Ok(current_value + value)
+                })?;
+            Ok(Response::new())
+        }
+
+        fn count(&self, ctx: QueryCtx<Self::QueryC>) -> Result<Self::CounterT, Self::Error> {
+            let count = self.counter.load(ctx.deps.storage)?;
+            Ok(count)
+        }
     }
 }
 
 // Making sure `Remote` can be stored in `#[cw_serde]` types
 #[cw_serde]
 #[allow(dead_code)]
-struct CustomStorage<'a, Contract> {
-    remote: sylvia::types::Remote<'a, Contract>,
-    interface_remote:
-        sylvia::types::Remote<'static, &'a dyn some_interface::SomeInterface<Error = StdError>>,
+pub struct ContractStorage<'a, Contract> {
+    remote: Remote<'a, Contract>,
```

Comment:
Having an example with the contract is good, but we still need working remotes for interfaces. It should not require reference.

Reply:
Fixed in #367. Will rebase after it is merged.